### PR TITLE
enh(notion connector): avoid long-running activities

### PR DESCRIPTION
**Pull request by fontanierh**
Instead of getting all databases' child pages from the "get to sync" activity, we now do this from the `notionSyncResultPageDatabaseWorkflow`, using one activity per result page of the "get children" notion endpoint.

The code will need some refactoring soon as it is becoming convoluted

### CODE REVIEW

File: connectors/src/connectors/notion/lib/notion_api.ts [hunk 1]
```diff
@@ -209,23 +210,6 @@ export async function getPagesAndDatabasesEditedSince(
         // through its pages.
         try {
           editedDbs[pageOrDb.id] = lastEditedTime;
-
-          // Note: we don't want to optimize this step due to Notion not always returning all the
```
Comment (fontanierh) [inline connectors/src/connectors/notion/lib/notion_api.ts:213]: moved this to a different workflow

**Review by spolu — COMMENTED** [review]
This all look reasonable but won't have time to review before I leave :/
Deferring to @lasryaric

File: connectors/src/connectors/notion/lib/notion_api.ts [hunk 2]
```diff
@@ -252,6 +236,74 @@ export async function getPagesAndDatabasesEditedSince(
...
+        { error: e },
+        "Error fetching result page from Notion API."
+      );
+      if (tries >= retry.retries) {
+        throw e;
+      }
+      const sleepTime = 500 * retry.backoffFactor ** tries;
+      tryLogger.info({ sleepTime }, "Sleeping before retrying.");
+      await new Promise((resolve) => setTimeout(resolve, sleepTime));
+      continue;
```
Comment (lasryaric) [inline connectors/src/connectors/notion/lib/notion_api.ts:300]: This continue here is a bit confusing. Can we remove it?

**Review by lasryaric — COMMENTED** [review]
I don't really understand this PR, it's complicated, but the code looks good to me.

**Review by lasryaric — CHANGES_REQUESTED** [review]
Just asking for one change about the `continue`.
Can't really comment as this PR is complicated to review but nothing else surprised me.

**Comment by fontanierh** [comment]
Done @lasryaric

**Pull request merged**